lib/matplotlib/tests/test_widgets.py

```diff
@@ -44,6 +44,36 @@ def test_rectangle_selector():
     check_rectangle(rectprops=dict(fill=True))
 
 
+@pytest.mark.parametrize('drag_from_anywhere, new_center',
+                         [[True, (100, 100)],
+                          [False, (50, 50)]])
+def test_rectangle_drag(drag_from_anywhere, new_center):
+    ax = get_ax()
+
+    def onselect(epress, erelease):
+        pass
+
+    tool = widgets.RectangleSelector(ax, onselect, interactive=True,
+                                     drag_from_anywhere=drag_from_anywhere)
+    # Create rectangle
+    do_event(tool, 'press', xdata=0, ydata=0, button=1)
+    do_event(tool, 'onmove', xdata=100, ydata=100, button=1)
+    do_event(tool, 'release', xdata=100, ydata=100, button=1)
+    assert tool.center == (50, 50)
+    # Drag inside rectangle, but away from centre handle to make sure rectangle
+    # is still moved, if *drag_from_anywhere* is enabled.
+    do_event(tool, 'press', xdata=25, ydata=25, button=1)
+    do_event(tool, 'onmove', xdata=75, ydata=75, button=1)
+    do_event(tool, 'release', xdata=75, ydata=75, button=1)
+    assert tool.center == new_center
+    # Check that in both cases, dragging outside the rectangle draws a new
+    # rectangle
+    do_event(tool, 'press', xdata=175, ydata=175, button=1)
+    do_event(tool, 'onmove', xdata=185, ydata=185, button=1)
+    do_event(tool, 'release', xdata=185, ydata=185, button=1)
+    assert tool.center == (180, 180)
```

Comment:
Generally, it is advisable to reduce the symmetry in tests, i.e. don't use the same values for x and y. That way, the test can catch an accidental mixup of the coordinates.

Reply:
👍